add warning about indirectly resolving scoped (#20302)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/dependency-injection.md

When using Entity Framework Core, the <xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.AddDbContext%2A> extension method registers `DbContext` types with a scoped lifetime by default.

> Do ***not*** resolve a scoped service from a singleton. It's also dangerous to resolve a scoped service from a singleton indirectly, for example, through a transient service. It may cause the service to have incorrect state when processing subsequent requests. It's fine to:

Reviewer: This looks fine as-is to me, but here's an alternative, _just 'cos_:

```suggestion
Do ***not*** resolve a scoped service from a singleton and be careful not to do so indirectly, for example, through a transient service. It may cause the service to have incorrect state when processing subsequent requests. It's fine to:
```
